### ISSUE TITLE
iOS toolchain broken with libimobiledevice 1.2.0 + Xcode 8

### ISSUE THREAD

**Issue by yjbanov** [flutter/flutter#5937]
## Steps to Reproduce

Upgrade to Xcode 8. Observe this issue: https://github.com/libimobiledevice/libimobiledevice/issues/356.

Workaround is available in comment https://github.com/libimobiledevice/libimobiledevice/issues/356#issuecomment-242553075 and I verified that it works.

/cc @chinmaygarde @tvolkert @cbracken

**Comment by dragostis** [comment]
This might get fixed in the next beta of Xcode?

**Comment by yjbanov** [comment]
I tried it on Xcode 8 final

**Comment by cbracken** [comment]
Running with with Xcode 8.0, latest ios-deploy (1.9.0), libimobiledevice (1.2.0) from brew, xcode command-line tools installed via `sudo xcode-select --install` against an iPhone 6S+ on iOS 10 worked for me.

If you can repro and dump version info for each of those, that'd help track this down.

**Comment by cbracken** [comment]
TL;DR an updated release of libimobiledevice on brew gets us rolling again. In the meantime a quick workaround is to install from head with homebrew:

```shell
brew uninstall libimobiledevice
brew install -s --HEAD libimobiledevice
```

Let's catch up with/send PRs to the libimobiledevice authors/brew maintainer to see if we can get a new release rolled out.

Some details:

*What works:* I'm able to build, install and run with the above setup on both the simulator and a real device.  

*What's broken:* so far, just that listing physical device name, e.g. in `flutter doctor`, fails. There may be other issues.

The issue occurs in `_getDeviceName()` in `lib/src/ios/devices.dart`. That method invokes `ideviceinfo` which fails with:
```shell
ERROR: Could not connect to lockdownd, error code -17
```

Running `dtruss` to trace syscalls suggests that this is the same issue as reported in the [thread](https://github.com/libimobiledevice/libimobiledevice/issues/356#issuecomment-242553075) you cite: we're failing due to permissions on `/var/db/lockdown/bunch_of_hex.plist`. `/var/db/lockdown` is permissioned 700 with user, group `_usbmuxd`.

As I say, the fixes are already landed upstream, looks like it's just a matter of getting a release rolled out on homebrew.

**Comment by cbracken** [comment]
Separate from the libimobiledevice issue is the fact that Xcode now wants a team specified to build and deploy to a physical device (as usual, no issue running on the simulator). Manually setting this in the Xcode project resolves the issue but we should see if there's an automated solution here.

**Comment by eseidelGoogle** [comment]
Is there something we can do in `flutter doctor` to detect this and help users?  FYI @danrubel

**Comment by cbracken** [comment]
As of this afternoon, we're down to just one remaining issue for Xcode 8, which is libimobiledevice version. Until a new version is released upstream we need users to install from head. Unfortunately the libimobiledevice tools don't include a `--version` option. 

We could potentially check if the user installed it via homebrew and check that the version there is > 1.2.0 or 1.2.0 at head. `brew info libimobiledevice` has the info. This doesn't help for those using MacPorts or hand-built versions, but it might catch a significant portion.

**Comment by cbracken** [comment]
A quick check for `flutter doctor` could be to run `idevicename` and check for exit code 0. This exits non-zero for the same reasons as other libidevice tools at older versions.

**Comment by cbracken** [comment]
An alternative to #6111 would be to eliminate our dependency on the libimobiledevice suite and replace it with command-line Xcode toolchain alternatives. Note that `ios_deploy`, which we use to install and launch, doesn't depend on libimobiledevice.

**Comment by eseidelGoogle** [comment]
Definitely a regression, but not really ours.  Removing regression keyword.

**Comment by eseidelGoogle** [comment]
@cbracken: My understanding is that we now check the necessary things and can close this?  Please re-open if I've misunderstood.

**Comment by cbracken** [comment]
Yep; all the necessary checks are in doctor now. Thanks for closing.
